test/e2e/network_segmentation.go

```diff
@@ -223,7 +232,8 @@ var _ = Describe("Network Segmentation", func() {
 						defaultNetNamespace := f.Namespace.Name + "-default"
 						_, err := cs.CoreV1().Namespaces().Create(context.Background(), &v1.Namespace{
 							ObjectMeta: metav1.ObjectMeta{
-								Name: defaultNetNamespace,
+								Name:   defaultNetNamespace,
+								Labels: map[string]string{RequiredUDNNamespaceLabel: ""},
```

Comment:
Commenting here but it related to other places as well.
Please consider introducing a helper to create the test namespace, it will reduce the noise in all places that create the namespace, and make it easier to introduce additional changes to such namespaces.
``` 
_, err := cs.CoreV1().Namespaces().Create(context.Background(), newTestNamespace(defaultNetNamespace))...
...
func newTestNamesapce(name string) *Namespace {
  return  &v1.Namespace{ ObjectMeta: metav1.ObjectMeta{
    Name: name,
    Labels: map[string]string{RequiredUDNNamespaceLabel: ""},
}}
```